src/main/java/com/example/medical_clinic/repository/PatientRepository.java

```diff
@@ -0,0 +1,32 @@
+package com.example.medical_clinic.repository;
+
+import com.example.medical_clinic.model.Patient;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+@RequiredArgsConstructor
+@Getter
+public class PatientRepository {
+    private final List<Patient> patients;
+
+    public Optional<Patient> findByEmail(String email) {
+        return patients.stream()
+                .filter(patient -> patient.getEmail().equalsIgnoreCase(email))
+                .findFirst();
+    }
+
+    public Patient addPatient(Patient patient) {
+        patients.add(patient);
+        return patient;
+    }
+
+    public Patient removePatient(Patient patient) {
+        patients.remove(patient);
```

Comment:
Ta metoda niezaleznie od tego czy został usuniety czy też nie zwraca zawsze to samo -> Pacjenta, którego dostała jako argument. 

ktoś usuwa pacjenta ktorego nie ma i i tak dostaje informacje ze sie udalo